src/Type/Constant/ConstantFloatType.php

```diff
@@ -45,18 +45,24 @@ public function equals(Type $type): bool
 		return $type instanceof self && abs($this->value - $type->value) < PHP_FLOAT_EPSILON;
 	}
 
+	private function castFloatToString(float $value): string
+	{
+		$precisionBackup = ini_get('precision');
```

Comment:
needs https://github.com/phpstan/phpstan/issues/3063 - how to solve it in this PR?

Reply:
fixed in https://github.com/phpstan/phpstan-src/pull/2362